app/controllers/master/api/providers_controller.rb

```diff
@@ -187,16 +187,16 @@ def plans
   end
 
   def get_partner
-    @partner = Partner.find_by_system_name(params[:partner])
+    @partner = Partner.find_by(system_name: params.permit(:partner)[:partner])
   end
 
   def get_application_plan
     application_plans = Account.master.application_plans.where(partner_id: @partner.try!(:id))
-    @application_plan = application_plans.find_by_system_name(params[:application_plan])
+    @application_plan = application_plans.find_by(system_name: params.permit(:application_plan)[:application_plan])
```

Comment:
permit seems not needed here as well

Reply:
I think it is OK. It will prevent some SQLi if `params[:application_plan]` is not a simple value but a Hash ;)
https://github.com/3scale/porta/blob/b0046173d5d11e13c48e7da9fceab7cdc7dd776b/config/initializers/rails_sqli.rb

Reply:
In Rails 5.1 it should not be treated as a Hash in case it is not permitted. We can test that.